Add support for MacOS-arm64 binary releases

JaimeSeqLabs opened the pull request:
# Description

Update the GA runner and JReleaser config to produce and upload Migtool GraalVM executables built for MacOS arm64 (M1, M2, M3, etc) processors.

> [!WARNING]  
> GA MacOS runners don't support running the migtool tests at the moment.

Migtool tests use Docker to spin up a MySQL database in which to run migrations. Currently, Docker does not work in arm64 GA runners because the nested virtualization is not supported for that family of processors :point_right: https://docs.github.com/en/actions/reference/runners/larger-runners#limitations-for-macos-larger-runners

> Nested-virtualization and Metal Performance Shaders (MPS) are not supported due to the limitation of Apple's Virtualization Framework

This may change in the future but, for now, I just skip the tests if running inside the MacOS runner (the tests still run in the Linux runner as usual).

JaimeSeqLabs committed:
feat: update GA and Jreleaser to build and upload MacOS-arm64
executables

JaimeSeqLabs committed:
fix: try setting up docker in macos GA runner

JaimeSeqLabs committed:
fix: skip tests if running inside a macos-arm64 GA runner

JaimeSeqLabs committed:
fix: skip native tests for macos-arm64 GA runner as well

munishchouhan commented:
build is working, tested here: https://github.com/munishchouhan/migtool/actions/runs/19079380505

The binary needs to be signed to make them work on mac, like its done for wave cli and tower cli
https://github.com/seqeralabs/wave-cli/blob/master/.github/workflows/build.yml

@jordigg can provide you with all the required information

<img width="613" height="291" alt="Screenshot 2025-11-04 at 20 02 00" src="https://github.com/user-attachments/assets/c75bc546-6f7d-48d2-8dc1-df251d710e7e" />

JaimeSeqLabs committed:
feat: separate builds for linux and macos, sign and notarize macos
binaries on release [release]

JaimeSeqLabs committed:
test: temporarily remove branch restrictions for build jobs, do dry-run
for jreleaser [release]

JaimeSeqLabs committed:
test: restore the branch restrictions and jrelease command

JaimeSeqLabs commented:
@munishchouhan, @jordigg  I refactored the build steps to separate linux and macos builds. I also added the same sign and notarize setup as the tw-cli repo but it looks like we are missing the MacOS certificate secrets for this repo (?)

I tried the sign step in https://github.com/seqeralabs/migtool/pull/57/commits/09dcf41ef2cb8e5c6ad6ec45c438ac723f7cd7b2 but the run failed -> https://github.com/seqeralabs/migtool/actions/runs/19145161087/job/54720726353

JaimeSeqLabs committed:
fix: rename the artifact paths for jreleaser [ci skip]

munishchouhan commented:
> @munishchouhan, @jordigg I refactored the build steps to separate linux and macos builds. I also added the same sign and notarize setup as the tw-cli repo but it looks like we are missing the MacOS certificate secrets for this repo (?)
> 
> I tried the sign step in [09dcf41](https://github.com/seqeralabs/migtool/commit/09dcf41ef2cb8e5c6ad6ec45c438ac723f7cd7b2) but the run failed -> https://github.com/seqeralabs/migtool/actions/runs/19145161087/job/54720726353

There are all org secrets, so it should work here too.
@jordigg anything we are missing?
I tried to login to seqera apple developer account but it requires MFA and code goes to some phone

munishchouhan commented:
We saw the same error in wave-cli too
https://github.com/seqeralabs/wave-cli/pull/20#issuecomment-1717530499

munishchouhan commented:
It may be an issue of org secrets present in this repo
checking with devops

munishchouhan committed:
dry run

munishchouhan committed:
Revert "dry run"

This reverts commit 3a7b3c89ebbd0bae0cc25af34283ca161777a8d1.

munishchouhan commented:
@JaimeSeqLabs codesign is working now, please go ahead with the pr
